### PR TITLE
refactor(core): pilot pure JAX programs with TF1D and PIC1D

### LINKED ISSUE #350: Define an explicit pure-JAX JaxProgram boundary for continuous and discrete systems
## Motivation

The current outer JIT wraps a bound method whose config, initial state, solver, and defaults are hidden closure state. Mutating that state can leave stale compiled constants or leak tracers. The custom Stepper also treats a “vector field” as a complete next-state map, while other solvers expose a true ODE derivative.

## Scope

Define an explicit numerical API, preferably composed with Equinox:

- JaxProgram(params, state, inputs, key) -> RawResult.
- ContinuousSystem.rhs(t, state, params, inputs) -> dstate_dt.
- DiscreteSystem.step(step, state, params, inputs, key) -> next_state.
- Appropriate Diffrax and scan/rollout execution adapters.
- A stable PyTree RawResult with final state, observations, and numerical status.

The program itself must cross the transform boundary as an explicit PyTree argument. Host-only configuration, units, tracking, plotting, paths, and clients remain outside.

## Acceptance criteria

- [ ] Program, parameters, state, runtime inputs, and PRNG key are explicit PyTrees at the transformed call.
- [ ] Numerical calls do not mutate Python objects, module fields, or input dictionaries.
- [ ] MLflow, filesystem, Pint, Pydantic, xarray, and matplotlib objects never enter the transformed boundary.
- [ ] Array-valued constants are visible PyTree leaves rather than hidden bound-method closure data.
- [ ] Repeated same-structure calls with changed state/inputs do not use stale values.
- [ ] Tests cover jit, grad, repeated calls, tracer-leak detection, and vmap where supported.
- [ ] One true continuous solver and one discrete-step solver achieve numerical parity with legacy paths.
- [ ] Random behavior is deterministic for a supplied key and changes for a different key.
- [ ] RawResult has a stable documented tree schema.
- [ ] Equinox is declared as a direct dependency if it becomes part of ADEPT’s public implementation contract.

## Dependencies

Coordinates with PreparedSimulation and explicit Params/State/Inputs contracts.

## Out of scope

- Wrapping host-side objects in eqx.Module.
- Logging or run creation inside JIT.
- Rewriting solver physics.
- Comprehensive distributed execution policy.

### LINKED ISSUE #353: Add logging-free SimulationSpec, SolverBuilder, PreparedSimulation, and solver registry
## Motivation

Preparation currently requires MLflow configuration, mutates an ADEPTModule through an ordered lifecycle, and selects solvers through a central if/elif chain. Tests, training loops, notebooks, and alternative experiment managers need deterministic preparation without tracking or artifact side effects.

## Scope

Introduce host-side contracts for:

- SimulationSpec: validated user intent and schema version.
- SolverBuilder.prepare(spec, *, key): deterministic construction.
- PreparedSimulation: explicit program, initial parameters/state/inputs, manifest, analyzer, and capabilities.
- RunManifest: raw/resolved configuration, units, seed/key provenance, code/dependency metadata, and a structural fingerprint.
- A public registry mapping stable solver names to builders, including caller-supplied builders.

Preparation should return immutable data rather than mutating an ordered lifecycle object. Tracking/run names do not belong in SimulationSpec.

## Acceptance criteria

- [ ] prepare() performs no MLflow calls, artifact writes, or process-global configuration changes.
- [ ] Preparation returns data instead of mutating a lifecycle object.
- [ ] Random initialization accepts an explicit key/seed and records reproducibility metadata.
- [ ] The same input spec/key produces an equivalent prepared structure.
- [ ] Input configurations remain reusable and are not mutated.
- [ ] Built-in and caller-supplied builders can register without editing a central switch.
- [ ] Unknown names and registration collisions produce actionable errors.
- [ ] Capabilities identify execution kind, dtype requirements, differentiability, batching, and supported placement.
- [ ] Existing YAML configurations can be adapted without changing physics meaning.
- [ ] Contract tests cover deterministic preparation, custom registration, and structural fingerprints.
- [ ] At least one existing solver uses the builder path.

## Dependencies

Coordinate the PreparedSimulation numerical payload with the pure-JAX program issue. The protocol can initially be implemented against a minimal placeholder program.

## Out of scope

- JIT, gradient, batching, or distributed implementation.
- MLflow lifecycle management.
- Replacing every solver-specific schema in one PR.
- Removing legacy lifecycle hooks.

### PR DESCRIPTION
**Pull request by joglekara**
## Summary

- add a stable `RawResult` plus explicit continuous `rhs` and discrete `step` contracts
- add Equinox/Diffrax and scan adapters whose transformed call is `program(params, state, inputs, key)`
- register logging-free builders for continuous TF1D and electrostatic PIC1D
- move runtime drivers out of hidden closure/config state and record deterministic key provenance
- keep `BaseTwoFluid1D`, `BasePIC1D`, `ADEPTModule`, and `ergoExo` intact as parallel fallback paths

## Migration boundary

This is an additive vertical slice. Existing entry points are not rerouted or deprecated. PIC transverse (`ey`) and stochastic forcing, plus TF learned trapping closures, fail early with an actionable message to use the legacy path.

The new tests exercise JIT, gradients, repeated same-structure inputs, tracer-leak detection, vmap, seeded preparation, import isolation, and non-equilibrium numerical parity with the legacy TF1D and PIC1D maps. They also verify that MLflow, Pint, Pydantic, xarray, and matplotlib objects do not enter prepared transform PyTrees.

## Validation

- `uv run --extra dev pytest -q tests/test_architecture tests/test_pic1d/test_bohm_gross.py tests/test_pic1d/test_two_stream.py` (40 passed)
- focused `uvx ty check` (passed)
- `uv run --extra dev pre-commit run --all-files` (passed)
- `uv run --extra docs sphinx-build -W --keep-going -b html docs/source /tmp/adept-docs-core-programs` (passed)

Closes #350
Closes #353
Refs #349
Refs #352

### CODE REVIEW

File: adept/_pic1d/helpers.py [hunk 1]
```diff
@@ -114,10 +112,11 @@ def _initialize_particles_(cfg: dict, simulation: PIC1DSimulation) -> dict:
         w_parts: list[np.ndarray] = []
 
         rng = np.random.default_rng(2025)  # fallback; per-subspecies seeds set below
-        for spec, sub_np, nprof in zip(sub_specs, np_per_sub, sub_profiles, strict=True):
+        for subspecies_index, (spec, sub_np, nprof) in enumerate(zip(sub_specs, np_per_sub, sub_profiles, strict=True)):
             sub_np = int(sub_np)
-            seed = spec.density_profile.noise_profile.noise_seed
-            rng = np.random.default_rng(int(seed))
+            configured_seed = spec.density_profile.noise_profile.noise_seed
+            this_seed = configured_seed if seed is None else seed + species_index * len(sub_specs) + subspecies_index
```
[the remark this answers is not in the document]
Comment (joglekara) [reply adept/_pic1d/helpers.py]: Fixed in 0cbbede. Explicit-key initialization now derives each stream with SeedSequence([seed, species_index, subspecies_index]), so variable component counts cannot collide. The (0, 0) stream remains byte-for-byte compatible with default_rng(seed), preserving legacy parity. Added a regression covering the reported (species 0, component 1) vs (species 1, component 0) collision, deterministic replay, and primary-stream compatibility.
